app/index.js

```diff
@@ -81,6 +81,7 @@ module.exports = class GeneratorLicense extends Generator {
         name: 'license',
         message: 'Which license do you want to use?',
         default: this.options.defaultLicense,
+        when: this.options.license == null,
```

Comment:
`in` operator in JavaScript only check for key match - not content.

You'll want `license.includes(this.options.license)`